src/gesture-handler/fabricUtils.ts

```diff
@@ -1,39 +1,84 @@
-import { NativeStackNavigatorProps } from '../native-stack/types';
+'use strict';
+/* eslint-disable */
 
-interface HostInstance {
-  _internalInstanceHandle: {
-    stateNode: {
-      node: Record<string, unknown>;
-    };
-  };
-  _nativeTag: number;
-}
+import { NativeStackNavigatorProps } from '../native-stack/types';
 
 type LocalGlobal = typeof global & Record<string, unknown>;
 
 export function isFabric() {
   return !!(global as LocalGlobal)._IS_FABRIC;
 }
 
-let findHostInstance: (ref: React.Component) => HostInstance | null = () => {
-  return null;
+export type ShadowNodeWrapper = {
+  __hostObjectShadowNodeWrapper: never;
 };
-if (isFabric()) {
-  try {
-    findHostInstance =
-      // eslint-disable-next-line @typescript-eslint/no-var-requires
-      require('react-native/Libraries/Renderer/shims/ReactFabric').findHostInstance_DEPRECATED;
-  } catch (e) {
-    throw new Error('[RNScreens] Cannot import `findHostInstance_DEPRECATED`.');
-  }
-}
 
+let findHostInstance_DEPRECATED: (ref: unknown) => void;
+
+let getInternalInstanceHandleFromPublicInstance: (ref: unknown) => {
+  stateNode: { node: unknown };
+};
+
+// Taken and modifies from reanimated
 export function getShadowNodeWrapperAndTagFromRef(
   ref: React.Ref<NativeStackNavigatorProps> | React.Component,
```

Comment:
This is bad here. We need to somehow decouple it from native-stack implementation or move the whole implementation to react-navigation. But that is story for another set of PRs.

Reply:
Till the native-stack v5 is in the repo we're fine, however we'll have to move it at some point.

Reply:
I'm talking about using `NativeStackNavigatorProps` here.

Reply:
I replaced it with generic Ref<React.Component> I don't think it's necessary to make it more precise as we're using it just to retrieve nativeTag and HostObject